staging/src/k8s.io/component-base/logs/json/json.go

```diff
@@ -0,0 +1,190 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package logs
+
+import (
+	"os"
+	"time"
+
+	"github.com/go-logr/logr"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+var (
+	JsonLogger logr.Logger
+	timeNow    = time.Now // Stubbed out for testing.
+)
+
+type noopInfoLogger struct{}
+
+func (l *noopInfoLogger) Enabled() bool                   { return false }
+func (l *noopInfoLogger) Info(_ string, _ ...interface{}) {}
+
+var disabledInfoLogger = &noopInfoLogger{}
+
+type infoLogger struct {
+	lvl zapcore.Level
+	l   *zap.Logger
+}
+
+var _ logr.InfoLogger = &infoLogger{}
+
+// Enabled always return true
+func (l *infoLogger) Enabled() bool { return true }
+
+// Info write message to error level log
+func (l *infoLogger) Info(msg string, keysAndVals ...interface{}) {
+	if checkedEntry := l.l.Check(l.lvl, msg); checkedEntry != nil {
+		checkedEntry.Time = timeNow()
+		checkedEntry.Write(handleFields(l.l, keysAndVals)...)
+	}
+}
+
+// handleFields converts a bunch of arbitrary key-value pairs into Zap fields.  It takes
+// additional pre-converted Zap fields, for use with automatically attached fields, like
+// `error`.
+func handleFields(l *zap.Logger, args []interface{}, additional ...zap.Field) []zap.Field {
+	// a slightly modified version of zap.SugaredLogger.sweetenFields
+	if len(args) == 0 {
+		// fast-return if we have no suggared fields.
+		return additional
+	}
+
+	// unlike Zap, we can be pretty sure users aren't passing structured
+	// fields (since logr has no concept of that), so guess that we need a
+	// little less space.
+	fields := make([]zap.Field, 0, len(args)/2+len(additional))
+	for i := 0; i < len(args); {
```

Comment:
```suggestion
	for i := 0; i < len(args) - 1; i+=2 {
```
If you apply other suggestion then it should look like this